lib/rucio/core/authentication.py

```diff
@@ -497,14 +542,22 @@ def __delete_expired_tokens_account(account, *, session: "Session"):
     :param account: Account to delete expired tokens.
     :param session: The database session in use.
     """
-    stmt_select = select(models.Token.token) \
-        .where(and_(models.Token.expired_at < datetime.datetime.utcnow(),
-                    models.Token.account == account)) \
-        .with_for_update(skip_locked=True)
-    tokens = session.execute(stmt_select).scalars().all()
-
-    for t in chunks(tokens, 100):
-        stmt_delete = delete(models.Token) \
-            .where(models.Token.token.in_(t)) \
-            .prefix_with("/*+ INDEX(TOKENS_ACCOUNT_EXPIRED_AT_IDX) */")
-        session.execute(stmt_delete)
+    select_query = select(
```

Comment:
Why not just perform a delete query directly ? Is this really so much easier on the database to explicitly lock the rows 100 at a time rather than leaving it to do however it desires ? But this is a comment for another change. No need to bundle it in this "sqlalchemy 2 syntax migration"

Reply:
Both to this and the previous comment, I assumed that the necessity for two queries (`SELECT` then `DELETE` as opposed to just `DELETE`) was to apply `FOR UPDATE SKIP LOCKED`. I opted for maintaining this implementation, without fully understanding the motivation behind it.

Reply:
Ah, yes, skip locked should have been the reason. This makes sense. Thanks for explaining